core/server/services/members/exporter/query.js

```diff
@@ -0,0 +1,88 @@
+const models = require('../../../models');
+const {knex} = require('../../../data/db');
+const moment = require('moment');
+
+module.exports = async function (options) {
+    const hasFilter = options.limit !== 'all' || options.filter || options.search;
+
+    let ids = null;
+    if (hasFilter) {
+        // do a very minimal query, only to fetch the ids of the filtered values
+        // should be quite fast
+        options.withRelated = [];
+        options.columns = ['id'];
+
+        const page = await models.Member.findPage(options);
+        ids = page.data.map(d => d.id);
+
+        /*
```

Comment:
Could you add a comment line explaining why this is commented out?

Reply:
Yeah, I tried last minute to improve the id fetching. But it didn't work reliably (I think filtering with search didn't work). I still think the commented code would have better performance if that is fixed because it 'escapes' from bookshelf. So I quickly commented it out before switching branches to work on the tests for https://github.com/TryGhost/Ghost/pull/14873 (switching between 4.x and 5.x takes a bit longer with the client/admin rename). Could have stashed the changes, but then I might forget about it 😬